lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -867,16 +866,20 @@ function dosomething_campaign_is_active($node) {
   */
  function dosomething_campaign_is_win_module($vars) {
   $hot_module_enabled = dosomething_helpers_get_variable('node', $vars['nid'], 'hot_module_enabled');
-  if (isset($vars['field_high_season']) && $hot_module_enabled) {
-     $today = dosomething_campaign_set_est_timezone(new DateTime());
-     $end_date = dosomething_campaign_set_est_timezone(new DateTime($vars['field_high_season'][0]['value2']), TRUE, TRUE);
-     $win_module_end = dosomething_campaign_set_est_timezone(date_modify(clone $end_date, '+30 days'), TRUE, TRUE);
-     $vars['hm_today'] = $today;
-     $vars['hm_end_date'] = $end_date;
-     // Win module is only enabled when variable is set, high season dates exist and there is still days left.
-     if (variable_get('dosomething_campaign_enable_win_module', NULL) == 1 && !empty($vars['field_high_season']) && (($end_date < $today)  && ($today < $win_module_end))) {
-       return TRUE;
-     }
+  $win_module_enabled = variable_get('dosomething_campaign_enable_win_module', NULL);
+  if ($win_module_enabled == 1 && isset($vars['field_high_season']) && $hot_module_enabled) {
+    $timing = dosomething_campaign_get_goals_times($vars);
+
+    // Set the values to the vars array, so we don't need to do the math again.
+    $vars['hm_today'] = $timing['today'];
+    $vars['hm_end_date'] = $timing['end_date'];
+
+    $win_module_end = dosomething_campaign_set_est_timezone(date_modify(clone $timing['end_date'], '+30 days'), TRUE, TRUE);
```

Comment:
i thought the goal was to remove this function (`dosomething_campaign_set_est_timezone`) since it caused wonkyness with the timezone?